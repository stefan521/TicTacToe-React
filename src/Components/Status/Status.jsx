// @flow
import React from 'react';

type Props = {
  status: {
    gameEnded: boolean,
    winner: string
  },
  xIsNext : boolean
};

function getStatus(status, xIsNext) {
  let statusLine;

  if (status.gameEnded && status.winner) {
    statusLine = `Winner: ${status.winner}`;
  } else if (status.gameEnded) {
    statusLine = 'Draw';
  } else {
    statusLine = `Next player: ${xIsNext === true ? 'X' : 'O'}`;
  }

  return statusLine;
}

function Status(props: Props) {
  const { status } = props;
  const { xIsNext } = props;

  return (
    <div className="game-status">
      {getStatus(status, xIsNext)}
    </div>
  );
}

export default Status;
