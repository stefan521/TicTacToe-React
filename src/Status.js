import React from 'react';

export default class Status extends React.Component {

  getStatus() {
    let status;
    if (this.props.winner) {
      status = 'Winner: ' + this.props.winner;
    } else {
      status = 'Next player: ' + (this.props.xIsNext ? 'X' : 'O');
    }

    return status;
  }

  render() {
    return (
      <div className="game-status">
        {this.getStatus()}
      </div>
    );
  }

}
