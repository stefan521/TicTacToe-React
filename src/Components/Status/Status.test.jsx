import React from 'react';
import { shallow } from 'enzyme';
import Status from './Status.jsx';

describe('Status', () => {
  it('displays a status', () => {
    const wrapper = shallow(<Status
      status={
        {
          gameEnded: false,
          winner: null,
        }
      }
      xIsNext={false}
    />);

    expect(wrapper.text()).toEqual('Next player: O');
    expect(wrapper).toMatchSnapshot();
  });

  it('displays a draw', () => {
    const wrapper = shallow(<Status
      status={
        {
          gameEnded: true,
          winner: null,
        }
      }
      xIsNext={false}
    />);

    expect(wrapper.text()).toEqual('Draw');
  });

  it('displays a win', () => {
    const wrapper = shallow(<Status
      status={
        {
          gameEnded: true,
          winner: 'O',
        }
      }
      xIsNext={true}
    />);

    expect(wrapper.text()).toEqual('Winner: O');
  });
});
