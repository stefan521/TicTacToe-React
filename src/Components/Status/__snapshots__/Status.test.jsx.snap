// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status displays a status 1`] = `
<div
  className="game-status"
>
  Next player: O
</div>
`;
